app-modules/service-management/tests/ServiceRequest/CreateServiceRequestTest.php

```diff
@@ -464,11 +465,74 @@
         ])
         ->create();
 
-    $serviceRequest = ServiceRequest::factory()->create([
+    $request = collect(CreateServiceRequestRequestFactory::new()->create([
         'priority_id' => ServiceRequestPriority::factory()->create([
             'type_id' => $serviceRequestTypesWithManager->getKey(),
         ])->getKey(),
-    ]);
+    ]));
+
+    livewire(CreateServiceRequest::class)
+        ->fillForm($request->toArray())
+        ->fillForm([
+            'respondent_id' => Contact::factory()->create()->getKey(),
+        ])
+        ->call('create');
+
+    $serviceRequest = ServiceRequest::first();
 
     expect($serviceRequest->assignments()->first())->user->id->toBe($user->getKey());
 });
+
+test('assignment type round robin will auto-assign to new service requests', function () {
+    asSuperAdmin();
+    $factoryUsers = User::factory()->licensed(LicenseType::cases())->count(3)->create();
+    $team = Team::factory()
+        ->hasAttached($factoryUsers, [], 'users')->create();
+
+    $serviceRequestTypeWithManager = ServiceRequestType::factory()
+        ->hasAttached(
+            factory: $team,
+            relationship: 'managers'
+        )
+        ->state([
+            'assignment_type' => ServiceRequestTypeAssignmentTypes::RoundRobin,
+        ])
+        ->create();
+
+    $request = collect(CreateServiceRequestRequestFactory::new()->create([
+        'priority_id' => ServiceRequestPriority::factory()->create([
+            'type_id' => $serviceRequestTypeWithManager->getKey(),
+        ])->getKey(),
+    ]));
+
+    $users = $team->users()->orderBy('name')->orderBy('id')->get();
+
+    foreach ($users as $user) {
+        livewire(CreateServiceRequest::class)
+            ->fillForm($request->toArray())
+            ->fillForm([
+                'respondent_id' => Contact::factory()->create()->getKey(),
+            ])
+            ->call('create')
+            ->assertHasNoFormErrors();
+
+        $latestServiceRequest = ServiceRequest::latest()->first();
+        $getServiceRequestType = ServiceRequestType::where('assignment_type', 'round-robin')->first();
+        expect($getServiceRequestType->assignment_type)->toBe(ServiceRequestTypeAssignmentTypes::RoundRobin);
+        expect($getServiceRequestType->fresh()->last_assigned_id)->ToBe($user->getKey());
```

Comment:
Does this fail of the `fresh()` is not used here? You just retrieved this from the database on line `520` and made no changes since. So that shouldn't be needed.